examples/pipeline.rs

```diff
@@ -0,0 +1,180 @@
+use std::cell::RefCell;
```

Comment:
I will leave it up to you whether you want to merge this example as is or spend some effort in merging more core infrastructure to glommio first.

I am fine with whatever decision you make!

However if this is to merged as an example as is, either temporarily or definitely, it would be nice to have more comments explaining to the reader what is the example doing, and how they can make use of a similar thing.

Examples are, after all, a way for people to learn.